server/src/main/java/org/opensearch/cluster/routing/allocation/ExistingShardsAllocator.java

```diff
@@ -60,6 +61,31 @@ public interface ExistingShardsAllocator {
         Setting.Property.PrivateIndex
     );
 
+    /**
+     * Boolean setting to enable/disable batch allocation of unassigned shards already existing on disk.
+     * This will allow sending all Unassigned Shards to the ExistingShard Allocator to  make decision to allocate
+     * in one or more go.
+     *
+     * Enable this setting if your ExistingShardAllocator is implementing the
+     * {@link ExistingShardsAllocator#allocateAllUnassignedShards(RoutingAllocation, boolean)} method.
+     * The default implementation of this method is not optimized and assigns shards one by one.
```

Comment:
By enabling this setting, GatewayAllocator will use batch implementation by default. In case of custom ExistingShardAllocator , the performance benefits will be visible if {@link ExistingShardsAllocator#allocateAllUnassignedShards(RoutingAllocation, boolean)} method is implemented efficiently for batch mode.

Reply:
No, if custom Allocator is set we wont be even running allocateAllUnassignedShards in AllocationService.

This was discussed [here](https://github.com/opensearch-project/OpenSearch/pull/8888/files/b1994fa1aff4789ae1ea125522a198c2e5970935#r1419024187)